doc/plugin/Turbo.md

```diff
@@ -0,0 +1,81 @@
+# Kaleidoscope-Turbo
+
+The `Turbo` plugin provides an extra key one can place on their keymap. While the
+key is pressed or toggled, pressing other keys will generate quick repeated
+inputs independent of the OS key repeation mechanics.
+
+## Using the plugin
+
+To use the plugin, simply include the header and enable the plugin and place
+`Key_Turbo` somewhere on your keymap. You may add additionally configure
+specific behaviors of the plugin as shown:
+
+```c++
+#include <Kaleidoscope.h>
+#include <Kaleidoscope-Turbo.h>
+#include <Kaleidoscope-LEDControl.h>
+
+// somewhere in the keymap...
+Key_Turbo
+
+KALEIDOSCOPE_INIT_PLUGINS(LEDControl, Turbo);
+
+void setup() {
+    Kaleidoscope.setup();
+
+    Turbo.interval = 30;
+    Turbo.toggle = true;
+    Turbo.flash = true;
+    Turbo.flash_interval = 80;
+    Turbo.activeColor = CRGB(0x64, 0x96, 0xed);
+}
+```
+
+## Plugin properties
+
+The `Turbo` object has the following user-configurable properties:
+
+### `.interval`
+
+> This property adjusts the timing between simulated keypresses. If you set this
+> too low, some programs might not like it.
+>
+> Defaults to 50
+
+### `.flash_interval`
+
+> This property adjusts the timing between the on/off states of the key LED.
+>
+> Defaults to 69
+
+### `.toggle`
+
+> This property changes the behavior of the turbo button. If you set it to true,
+> pushing the button will toggle between states.
```

Comment:
It's not exactly clear to me on a first read what the `false` state here does. If I understand it right, then by default, will only do its thing while held. With `.toggle` set, it would do its thing when tapped, until tapped off a second time. Is that correct?

In that case, I think we should be calling it `.sticky()`, modeled after the OneShot naming. "Toggle" to me suggests it toggles the Turbo behaviour itself on and off, which doesn't make much sense.

For `.sticky()`, I'd use something like the following as documentation:

```markdown
### `.sticky([bool])`

> This method makes the Turbo functionality sticky, so it remains in effect not only while
> it is held, but after it is released too, until it is toggled off with another tap. Without
> arguments, the method enables the sticky functionality. Passing a boolean argument
> sets stickiness to the given value.
>
> Defaults to `false`.
```